src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTest.php

```diff
@@ -37,10 +41,99 @@ public function testForward()
         $container->expects($this->at(0))->method('get')->will($this->returnValue($requestStack));
         $container->expects($this->at(1))->method('get')->will($this->returnValue($kernel));
 
-        $controller = new Controller();
+        $controller = new TestController();
```

Comment:
Even if I agree with you that "non-easily-testable" code is a smell, in this specific case, this is perfectly valid IMO.

Please, don't overcomplicate things, this class is how we tie everything together, easy and simple.